lib/MusicBrainz/Server/Report/ArtistsThatMayBeGroups.pm

```diff
@@ -4,16 +4,45 @@ use Moose;
 with 'MusicBrainz::Server::Report::ArtistReport',
      'MusicBrainz::Server::Report::FilterForEditor::ArtistID';
 
-sub query {
-    q{SELECT DISTINCT ON (artist.id) artist.id AS artist_id,
-       row_number() OVER (ORDER BY artist.name COLLATE musicbrainz, artist.id)
-     FROM artist
-     JOIN l_artist_artist ON l_artist_artist.entity1=artist.id
-     JOIN link ON link.id=l_artist_artist.link
-     JOIN link_type ON link_type.id=link.link_type
-     WHERE (artist.type NOT IN (2, 5, 6) OR artist.type IS NULL)
-       AND link_type.name IN ('collaboration', 'member of band', 'conductor position')};
-}
+sub query {<<~'SQL'}
+  WITH possible_groups_entity0 AS (
+         SELECT artist.id, artist.name
+         FROM artist
+         JOIN l_artist_artist laa ON laa.entity0 = artist.id
+         JOIN link ON link.id = laa.link
+         WHERE (
+            artist.type NOT IN (2, 5, 6) -- group, orchestra, choir
+            OR
+            artist.type IS NULL
+         )
+         AND link.link_type IN (722)), -- subgroup
+       possible_groups_entity1 AS (
+         SELECT artist.id, artist.name
+         FROM artist
+         JOIN l_artist_artist laa ON laa.entity1 = artist.id
+         JOIN link ON link.id = laa.link
+         JOIN link_type ON link_type.id = link.link_type
+         WHERE (
+            artist.type NOT IN (2, 5, 6) -- group, orchestra, choir
+            OR
+            artist.type IS NULL
+         )
+         AND link.link_type IN (
+            103, -- member of band
+            102, -- collaboration
+            305, -- conductor position
+            965, -- artistic director
+            855, -- composer-in-residence
+            722  -- subgroup
+         )
```

Comment:
Prior to the latest force-push, you just had `AND link_type.name IN ('member of band', 'collaboration', 'conductor position', 'subgroup'))` here. Was adding "artistic director" and "composer-in-residence" intentional?